File: src/main/Leetcode00027.java
import java.util.*;

public class Leetcode00027 {
    public int removeElement(int[] nums, int val) {
        int l = 0, r = 0;
        while (r < nums.length) {
            if (nums[r] != val) {
                nums[l++] = nums[r];
            }
            r++;
        }
        return l;
    }
}

File: src/main/Leetcode00027.swift
class Solution {
    func removeElement(inout nums: [Int], _ val: Int) -> Int {
        var l = 0
        var r = 0
        while r < nums.count {
            if nums[r] != val {
                if l != r {
                    swap(&nums[l], &nums[r])
                }
                l += 1
            }
            r += 1
        }
        return l
    }
}
